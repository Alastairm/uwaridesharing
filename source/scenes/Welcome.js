import React, { Component, Proptypes} from 'react';
import { AsyncStorage, Image, StyleSheet, Text, View } from 'react-native';
import Styles from './Styles.js';

import UserForm from './UserForm.js';
import Map from './Map.js';
import Button from '../components/Button.js';

const backgroundImage = {uri: 'https://fleeteng-static.s3.amazonaws.com/assets/background-081a9027c8c821541bf2d7816138fc87.jpg'};

export default class Welcome extends Component{
  constructor(props) {
    super(props);
    this.onNext = this.onNext.bind(this);
    this.onClear = this.onClear.bind(this);
  }
  async onNext() {
    try{
      let userSaved = await AsyncStorage.getItem('user.saved');
      if (userSaved == 'true') {
        this.props.navigator.push({
          component: Map
        });
      } else {
        this.props.navigator.push({
          component: UserForm
        });
      }
    } catch(error) {
      this.props.navigator.push({
        component: UserForm
      });
    }
  }
  async onClear() {
    await AsyncStorage.clear();
  }
  render() {
    return(
      <View style={Styles.scene}>
        <Image
        source={backgroundImage}
        style={Styles.backgroundImage}
        resizeMode={Image.resizeMode.sretch}>
        <View style={Styles.scene}>
          <View style={{alignSelf: 'center'}}>
            <Text style={Styles.welcomeTitle}>
              UniRide
            </Text>
            <Text style={Styles.welcomeSubtitle}>
              UWA Rideshare
            </Text>
          </View>
          <Button
            onPress={this.onClear}>
            Clear App Data
          </Button>
          <Button
            onPress={this.onNext}>
            Hello
          </Button>
          <Button
            onPress={this.onNext}>
            Get Started
          </Button>
          </View>
      </Image>
    </View>
    );
  }
}
