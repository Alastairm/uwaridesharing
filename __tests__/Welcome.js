import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Welcome from '../source/scenes/Welcome.js';

require('firebase');


test('Matches snapshot', () => {
  const render = renderer.create(
    <Welcome />,
  ).toJSON();
  expect(render).toMatchSnapshot();
});
