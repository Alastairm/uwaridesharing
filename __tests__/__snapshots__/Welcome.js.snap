// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Image
    resizeMode={undefined}
    source={
      Object {
        "uri": "https://fleeteng-static.s3.amazonaws.com/assets/background-081a9027c8c821541bf2d7816138fc87.jpg",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": null,
        "width": null,
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "navy",
              "fontSize": 108,
              "textAlign": "center",
            }
          }
        >
          UniRide
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "navy",
              "fontSize": 24,
              "textAlign": "center",
            }
          }
        >
          UWA Rideshare
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 60,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#faba12",
              "borderRadius": 4,
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "opacity": 1,
              "overflow": "hidden",
              "width": 180,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                null,
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                null,
              ]
            }
          >
            Sign In
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 60,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#303030",
              "borderRadius": 4,
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "opacity": 1,
              "overflow": "hidden",
              "width": 180,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                null,
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                null,
              ]
            }
          >
            API Test
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 60,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#303030",
              "borderRadius": 4,
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "opacity": 1,
              "overflow": "hidden",
              "width": 180,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                null,
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                null,
              ]
            }
          >
            Clear App Data
          </Text>
        </View>
      </View>
    </View>
  </Image>
</View>
`;
